commands_to_load/SFU.py

```diff
@@ -0,0 +1,249 @@
+from discord.ext import commands
+import logging
+import time
+import json # dont need since requests has built in json encoding and decoding
+import requests as req
+import re
+from helper_files.embed import embed 
+import helper_files.settings as settings
+import html
+
+logger = logging.getLogger('wall_e')
+sfuRed = 0xA6192E
+
+class SFU():
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def sfu(self, ctx, *course):
+        logger.info('[SFU sfu()] sfu command detected from user ' + str(ctx.message.author))
+        logger.info('[SFU sfu()] arguments given: ' + str(course))
+
+        if(not course):
+            eObj = embed(title='Missing Arguments', author=settings.BOT_NAME, avatar=settings.BOT_AVATAR, colour=sfuRed, content=[['Usage', '`.sfu <arg>`'],['Example', '`.sfu cmpt300`']], footer='SFU Error')
+            await ctx.send(embed=eObj)
+            logger.info('[SFU sfu()] missing arguments, command ended')
+            return
+
+        year = time.localtime()[0]
+        term = time.localtime()[1]
+
+        if(term <= 4):
+            term = 'spring'
+        elif(term >= 5 and term <= 8):
+            term = 'summer'
+        else:
+            term = 'fall'
+        
+        # Check if arg needs to be manually split
+        if(len(course) == 1):
+            #split
+            crs = re.findall('(\d*\D+)', course[0])
+            if(len(crs) < 2):
+                crs = re.split('(\d+)', course[0])
+
+            if(len(crs) < 2):
+                # Bad args
+                eObj = embed(title='Bad Arguments', author=settings.BOT_NAME, avatar=settings.BOT_AVATAR, colour=sfuRed, content=[['Usage', '`.sfu <arg>`'],['Example', '`.sfu cmpt300`']], footer='SFU Error')
+                await ctx.send(embed=eObj)
+                logger.info('[SFU outline()] bad arguments, command ended')
+                return
+            
+            courseCode = crs[0]
+            courseNum = crs[1]
+        else:
+            courseCode = course[0]
+            courseNum = course[1]
+
+        url = 'http://www.sfu.ca/bin/wcm/academic-calendar?%s/%s/courses/%s/%s' % (year, term, courseCode, courseNum)
+        logger.info('[SFU sfu()] url for get request constructed: %s' % url)
+
+        res = req.get(url)
+        if(res.status_code == 200):
+            logger.info('[SFU sfu()] get request successful')
+            data = res.json()
+        else:
+            logger.info('[SFU sfu()] get resulted in 404')

```

Comment:
can you guarantee that the error will always be 404?

Reply:
done. changed for both outline and sfu commands

Reply:
you sure? cause its still in the logger.info....